Refactor/update speakers functionality

shrinkray opened the pull request:
# Pull Request

## Description

<!-- Provide a brief description of the changes in this PR -->

## Type of Change

<!-- Mark the relevant option with an "x" -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactoring
- [ ] Performance improvement
- [x] Accessibility improvement
- [ ] Other (please describe):

## Related Issues

<!-- Link any related issues using "Fixes #123" or "Closes #123" -->
Fixes #105 

## Code Quality Checklist

<!-- Mark completed items with an "x" -->

- [x] Code follows the project's coding standards
- [x] Self-review of the code has been performed
- [ ] Code is properly commented where necessary
- [ ] No console.log statements left in the code
- [ ] No hardcoded values that should be configurable
- [ ] Error handling is implemented where appropriate

## Testing

<!-- Mark completed items with an "x" -->

- [x] Changes have been tested locally
- [ ] All existing tests pass (if applicable)
- [ ] New tests have been added for new functionality (if applicable)
- [ ] Manual testing has been performed
- [x] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

<!-- Mark completed items with an "x" -->

- [x] All interactive elements are keyboard accessible
- [ ] Color contrast meets WCAG AA standards
- [x] Images have appropriate alt text
- [ ] Form labels are properly associated
- [x] Focus indicators are visible
- [ ] Screen reader compatibility has been tested

## Build & Deployment

<!-- Mark completed items with an "x" -->

- [x] Application builds successfully
- [x] No build warnings or errors
- [ ] Dependencies are up to date
- [x] No security vulnerabilities introduced

## Screenshots (if applicable)

<!-- Add screenshots to help explain your changes -->

### Updated Hackathon location in LandingSection

<img width="706" height="165" alt="Screenshot 2025-11-06 at 12 57 12 PM" src="https://github.com/user-attachments/assets/1b8f3d91-a51f-4c76-93b7-a7d8f23a4a7c" />

### Hackathon Schedule Display

<img width="1170" height="826" alt="Screenshot 2025-11-06 at 12 57 07 PM" src="https://github.com/user-attachments/assets/dc26a35a-5a66-4199-8d2a-ef710a3c3dcb" />

## Additional Notes

<!-- Any additional information that reviewers should know -->

- I added items requested in whatsapp for Koziol
- Added sorting feature for dev team (similar to speakers)
- Refactor code on several files for accessibility as identified in the commit

---

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.

shrinkray committed:
feat: add dave likeness

-two images in speaker folder

shrinkray committed:
refactor: update speaker.js

-move Nilesh to build with AI

-add Dave's talk

shrinkray committed:
refactor: expand speaker section by default

shrinkray committed:
refactor: add method to sort team

-use first name, same as speakers

shrinkray committed:
refactor: accessibility updates

add aria-hidden true to decorative images

-add alt-tags to avatars

-add focus styles

add descriptive aria-label attr, descriptions

-other description alt text

-add visible focus indicators on buttons

-open in new tab warning

-modal nav have focus

shrinkray committed:
refactor: fix typo in speaker description

ShugKnight24 committed:
fix: upate hackathon location in LandingSection

ShugKnight24 committed:
feat: display Hackathon schedule in the SessionSection

ShugKnight24 commented:
Updated Hackathon Location in LandingSection

<img width="706" height="165" alt="Screenshot 2025-11-06 at 12 57 12 PM" src="https://github.com/user-attachments/assets/4f2cc011-2773-49c6-9ae4-8318584f84b7" />


Display Hackathon Schedule in SessionsSection

<img width="1170" height="826" alt="Screenshot 2025-11-06 at 12 57 07 PM" src="https://github.com/user-attachments/assets/09eb7dcd-e44d-43cd-a645-c4ccc3d8cba7" />

ShugKnight24 committed:
refactor: update Dave's image that is used by the SpeakerCard

ShugKnight24 committed:
refactor: remove `useMemo`, team data is static

- The dependency is static and won't change

shrinkray committed:
refactor: replace Dave's photo

-crop image, remove bkg, clone

ShugKnight24 committed:
fix: address copilot review

ShugKnight24 committed:
refactor: abstract no sessions / no speakers message

- Center its display when it's shown

ShugKnight24 commented on src/components/sessions/HackathonSchedule.jsx:
```diff
@@ -0,0 +1,59 @@
+          <div className="space-y-8">
+            {hackathonSchedule.map((item, index) => (
+              <div
+                key={index}
```
resolved by [7c6307d](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/7c6307d8366a91c62dfa0ba9e5829902175e249f)

ShugKnight24 commented on src/layouts/SessionsSection.jsx:
```diff
@@ -217,7 +218,9 @@ const SessionsSection = ({
-          {combinedSpeakerData && combinedSpeakerData.length ? (
+          {currentSession === 'Hackathon' ? (
+            <HackathonSchedule />
+          ) : combinedSpeakerData && combinedSpeakerData.length ? (
```
resolved by [7c6307d](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/7c6307d8366a91c62dfa0ba9e5829902175e249f)

ShugKnight24 commented on src/components/sessions/HackathonSchedule.jsx:
```diff
@@ -0,0 +1,59 @@
+const hackathonSchedule = [
+  { time: '8:00 AM', title: 'Check-In & Breakfast' },
+  { time: '9:00 AM', title: 'Morning Meditation & Yoga' },
+  { time: '9:00 AM', title: 'AI Hackathon Keynote' },
```
I'd prefer these be two separate entries though I can display them left to right on the same node if that's better

ShugKnight24 replied:
displaying events at a single time side by side - resolved by [3789178](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/378917851917cee7cead928b8d88206216e38082)

ShugKnight24 committed:
refactor: correct schedule & display multiple events at single time

- Items would alternate. Updated so if there are two things at a single time both will appear

ShugKnight24 committed:
refactor: add descriptions to event nodes that appear on hover

ShugKnight24 commented:
Updates

<img width="1600" height="348" alt="Screenshot 2025-11-06 at 3 32 59 PM" src="https://github.com/user-attachments/assets/fbc6d1d6-2add-4348-ad89-5a5b3cbdbdff" />
<img width="1128" height="820" alt="Screenshot 2025-11-06 at 5 03 27 PM" src="https://github.com/user-attachments/assets/ee960f76-6da2-4f11-b483-ed39a73cff72" />
<img width="543" height="258" alt="Screenshot 2025-11-06 at 5 03 32 PM" src="https://github.com/user-attachments/assets/47d01d67-fac5-4d71-8375-1de34dea7ed7" />
<img width="1214" height="671" alt="Screenshot 2025-11-06 at 5 03 38 PM" src="https://github.com/user-attachments/assets/87fef3cf-f302-4180-9084-f4c42af10d7a" />

shrinkray commented:
I asked CoPilot to review then maybe I will be able to review

ShugKnight24 committed:
refactor: abstract out event card to DRY the code

ShugKnight24 committed:
fix: address copilot review

ShugKnight24 commented on src/components/sessions/HackathonSchedule.jsx:
```diff
@@ -0,0 +1,172 @@
+                          {/* Arrow */}
+                          <div className="absolute left-1/2 top-full -mt-1 size-0 -translate-x-1/2 border-x-8 border-t-8 border-x-transparent border-t-gray-900" />
+                        </div>
+                      </div>
```
Didn't make it a button - removed `cursor-pointer`

Resolved by [02bc9f5](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/02bc9f519b456e1f26566c80cd0af53b4c2b744b)

ShugKnight24 commented on src/components/sessions/HackathonSchedule.jsx:
```diff
@@ -0,0 +1,172 @@
+                          {/* Arrow */}
+                          <div className="absolute left-1/2 top-full -mt-1 size-0 -translate-x-1/2 border-x-8 border-t-8 border-x-transparent border-t-gray-900" />
+                        </div>
+                      </div>
```
use optional chaining

resolved by [02bc9f5](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/02bc9f519b456e1f26566c80cd0af53b4c2b744b)

ShugKnight24 commented on src/layouts/SessionsSection.jsx:
```diff
@@ -96,6 +97,24 @@ const SessionsSection = ({
+      <a
+        href="https://www.papercall.io/midevfest2025"
+        target="_blank"
+        className="flex items-center rounded bg-sky-900 px-8 py-5 text-primary-50 shadow-xl transition delay-75 duration-100 ease-in-out hover:-translate-y-1 hover:scale-110 hover:cursor-pointer"
```
resolved by [02bc9f5](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/02bc9f519b456e1f26566c80cd0af53b4c2b744b)

ShugKnight24 commented on src/layouts/SessionsSection.jsx:
```diff
@@ -96,6 +97,24 @@ const SessionsSection = ({
+        href="https://www.papercall.io/midevfest2025"
+        target="_blank"
+        className="flex items-center rounded bg-sky-900 px-8 py-5 text-primary-50 shadow-xl transition delay-75 duration-100 ease-in-out hover:-translate-y-1 hover:scale-110 hover:cursor-pointer"
+        rel="noreferrer"
```
this shouldn't be seen anymore and the papercall is no longer valid - the point still stands

For future papercalls

resolved by [02bc9f5](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/02bc9f519b456e1f26566c80cd0af53b4c2b744b)

ShugKnight24 commented on src/data/2025/speakers.js:
```diff
@@ -873,4 +874,28 @@ export const SpeakersData = [
+    bio: `GDG-A2 & Detroit Co-Organizer, and Director, Software Engineering IT Innovation & Back of House at Little Caesars. Prior to joining Little Caesars Dave spent over 25 years in the Software Consulting world. His teams have submitted over 250 apps to the app stores, but he now leads teams focused on web development, embedded development and software administration. He's been helping run co ...
+    twitter: 'DaveKoziol',
+    linkedIn: 'https://www.linkedin.com/in/davekoziol/',
+    url: 'https://www.autoworkz.org/',
```
resolved by [02bc9f5](https://github.com/GDG-Detroit/devfest-website/pull/106/commits/02bc9f519b456e1f26566c80cd0af53b4c2b744b)

ShugKnight24 committed:
fix: prevent background scrolling when SpeakerDetails is open

shrinkray reviewed (commented):
Still not allowed to approve but from what I see, it is delivering more for accessibility.

ShugKnight24 committed:
refactor: remove speaker who can't make the event

shrinkray commented:
OK thanks @JustinBeaudry I've got another panel to add. I was confused on why I was unable to add a review here. Thanks everyone!
